src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs

```diff
@@ -1428,34 +1470,41 @@ private void LoadAndValidateNetModuleAttributes(ref CustomAttributesBag<CSharpAt
 
                 if (Interlocked.CompareExchange(ref lazyNetModuleAttributesBag, netModuleAttributesBag, null) == null)
                 {
-                    this.AddDeclarationDiagnostics(diagnostics);
+                    if (!noCache)
```

Comment:
Does this mean if we make a call with `noCache: true` and later with `noCache: false`, we could lose diagnostics?

Reply:
The call with `noCache: true` doesn't save the attribute bag into the symbol. So when we try to ForceComplete on the symbol, we'll have a call with `noCache: false` that computes/saves the attribute bag and the diagnostics too.